LiskHQ/lisk-hub
# Setup React error boundaries to prevent one component breaking the whole app - Closes #643

### What was the problem?
- There is no implementation to prevent not rendering the component tree when an error in a leaf component occurred. 

### How did I fix it?
- Wrap every Route component in an error boundary
- Provide reload button
- Provide send mail to support with stack trace and app version number
### How to test it?
- It's difficult to trigger an exception from the browser console inside the component logic, therefore a separate branch has been provided to test how the app behaves when theres an error in one of the leaf components. you can check it here https://jenkins.lisk.io/test/lisk-hub/643-setup-react-error-boundaries-test-with-error/#/dashboard
- please test that the rest of application is usable, while the dashboard shows the errorBoundary instead.

### Review checklist
- The PR solves #643
- All new code is covered with unit tests
- All new features are covered with e2e tests
- All new code follows best practices

![screen shot 2018-05-03 at 10 46 54](https://user-images.githubusercontent.com/668295/39568333-262d0b42-4ec2-11e8-99ee-49c56338fda2.png)
<img width="988" alt="screen shot 2018-05-03 at 10 47 52" src="https://user-images.githubusercontent.com/668295/39568334-264dff8c-4ec2-11e8-90bf-3f69420c6667.png">
![screen shot 2018-05-03 at 10 48 16](https://user-images.githubusercontent.com/668295/39568335-266a5d80-4ec2-11e8-90d6-340f1f39318d.png)
